themes/bootstrap3/templates/search/results.phtml

```diff
@@ -1,8 +1,26 @@
 <?php
   // Set up page title:
   $lookfor = $this->results->getUrlQuery()->isQuerySuppressed() ? '' : $this->params->getDisplayQuery();
+  
+  if (isset($this->parseError)):
+    $headTitleText = $this->transEsc('nohit_parse_error');
+  elseif ($recordTotal < 1):
+    $headTitleText = $this->translate('nohit_lookfor_html', ['%%lookfor%%' => $this->escapeHtml($lookfor)]);
+  else:
+    $transParams = [
```

Comment:
I think you can likely reduce redundancy here by replacing this with:

```
$showingHtml = $this->context()->renderInContext('search/controls/showing.phtml', ['lookfor' => $lookfor, 'recordTotal' => $recordTotal]);
$headTitleText = $this->layout()->srmessage;
```

...and then you can replace the call to renderInContext below with an output of the `$showingHtml` variable captured here. It would probably be good to initialize $showingHtml to an empty string in the other cases, just to be sure there are no weird code paths that reference an undefined variable.

This isn't exactly ideal, as it has a bit of spaghetti-code flavor to it, but I think it's probably preferable to duplicating the logic.

There's also no law that says we have to call `$this->headTitle` at the top of the template (though we almost always do). Maybe if we want to keep all the parts closer together, we could assign $headTitle a default value at the top of the template, then assign the "showing" text down below in the scenario where we are rendering the template, and then we can call the headTitle helper at the very bottom of the template.

Does that help and/or make any sense? :-)

Reply:
Thanks, @demiankatz! That was really helpful. It looks like I still need to run both strip_tags() and html_entity_decode() on $this->layout()->srmessage, but I like this much better.

Reply:
That's great, @stephanieleary, I'm glad I could help. Do you plan to push updates to this PR? I'll look forward to seeing them when they're ready. :-)

Reply:
Yes! Sorry about the delay; I've been moving to a new dev environment.